extensions/amp-bind/0.1/amp-state.js

```diff
@@ -62,6 +62,12 @@ export class AmpState extends AMP.BaseElement {
 
   /** @override */
   mutatedAttributesCallback(mutations) {
+    const viewer = viewerForDoc(this.getAmpDoc());
+    if (!viewer.isVisible()) {
+      const TAG = this.getName_();
+      dev().error(TAG, 'Viewer must be visible before mutation.');
+      return;
```

Comment:
I considered that but I think this is a serious enough bug that we should throw an error vs. silently delay execution.